File: cpe49題/10035 Primary Arithmetic.cpp
#include <iostream>
using namespace std;

int main() {
    int n,m;
    while(cin>>n>>m){
    	if(n==0 && m==0)break;
    	int result=0;
    	int add=0;
    	int tmp=0;
    	int sum=0;
    	while(n>0||m>0){
    		tmp=n%10+m%10;
    		sum=tmp+add;
	    	if(sum>=10){
	    		result+=1;
	    		add=sum/10;
	    	}else{
	    		add=0;
	    	}
	    	n/=10;
			m/=10;
    	}
		if(result==0){
			cout<<"No carry operation.";
		}else if(result==1){
			cout<<result<<" carry operation.";
		}else{
			cout<<result<<" carry operations.";
		}
		cout<<endl;
    }
}

File: cpe49題/10035 Primary Arithmetic.py
while True:
	n1,n2=map(int,input().split())
	if n1==0 and n2==0:
		break
	carry=0
	add=0
	
	while n1>0 or n2>0:
		tmp=n1%10+n2%10+add
		if tmp>=10:
			add=1
			carry+=1
		else:
			add=0
		n1//=10
		n2//=10
	
	if carry==1:
		print(f"{carry} carry operation.")
	elif carry:
		print(f"{carry} carry operations.")
	else:
		print("No carry operation.")
